main.go

```diff
@@ -1,70 +1,133 @@
 package main
 
 import (
-	"bytes"
-	rand "crypto/rand"
-	"crypto/sha256"
 	"crypto/tls"
-	"encoding/base64"
+	"crypto/x509"
 	"encoding/json"
-	"flag"
 	"fmt"
-	"github.com/cloudentity/sample-go-mtls-oauth-client/pkg/acp"
-	"github.com/gorilla/securecookie"
-	"io"
 	"log"
+	"net"
 	"net/http"
-	"strconv"
-)
+	"net/url"
+	"os"
+	"path/filepath"
+	"runtime"
+	"text/template"
+	"time"
 
-var (
-	acpOAuthConfig acp.Config
-	clientID       = flag.String("clientId", "", "Application client ID")
-	issuerURL      = flag.String("issuerUrl", "https://localhost:8443/default/default", "Issuer URL with provided tenant, and server ID")
-	port           = flag.String("port", "18888", "Port where callback, and login endpoints will be exposed")
-	host           = flag.String("host", "localhost", "Host where your client applications is running")
-	redirectHost   = flag.String("redirectHost", "localhost", "Host where the OAuth Server will redirect to")
-	certPath       = flag.String("cert", "certs/cert.pem", "A path to the file with a certificate")
-	keyPath        = flag.String("key", "certs/cert-key.pem", "A path to the file with a private key")
-	serverCertPath = flag.String("serverCert", "certs/server-cert.pem", "A path to the file with a server certificate")
-	pkceEnabled    = flag.Bool("pkce", false, "Enables PKCE flow")
-
-	secureCookie = securecookie.New(securecookie.GenerateRandomKey(64), securecookie.GenerateRandomKey(32))
+	"github.com/caarlos0/env"
+	acp "github.com/cloudentity/acp-client-go"
+	"github.com/dgrijalva/jwt-go"
 )
 
-const challengeLength = 43
+// in-memory token test store
+var token acp.Token
+var templ *template.Template
+var usePyron bool
+var xsslCertHash string
+var resourceURL string
+var injectCertMode bool
+
+type Config struct {
+	ClientID          string `env:"CLIENT_ID,required"`
+	CertPath          string `env:"CERT_PATH,required"`
+	KeyPath           string `env:"KEY_PATH,required"`
+	RootCA            string `env:"ROOT_CA,required"`
+	PORT              int    `env:"PORT,required"`
+	RedirectHost      string `env:"REDIRECT_HOST,required"`
+	WellKnown         string `env:"WELL_KNOWN_URL,required"`
+	WellKnownURL      *url.URL
+	IssuerURL         *url.URL `env:"ISSUER_URL"`
```

Comment:
If issuer_url is read from the well-known, remove the env tag here and in other config params that are set programmatically.